src/replication.c

```diff
@@ -831,32 +1024,78 @@ ask_replica_status_all(spec_t *spec)
 	MTX_UNLOCK(&spec->rq_mtx);
 }
 
-/*
- * This function processes the status response received from replica
- * and update spec's healthy/degrade count
- */
+static void
+update_rebuild_info(spec_t *spec, mgmt_ack_t *ack_data)
+{
+
+	uint64_t data_len = 0;
+
+	MTX_LOCK(&spec->rq_mtx);
+
+	if (spec->rebuild_info == NULL)
+		spec->rebuild_info = (mgmt_ack_t *)malloc(sizeof (mgmt_ack_t) *
+		    spec->rebuild_replica_count);
+
+	memcpy(&spec->rebuild_info[spec->rebuild_info_ack_recvd], ack_data,
+	    sizeof (mgmt_ack_t));
+	spec->rebuild_info_ack_recvd++;
+
+	if (spec->rebuild_info_ack_recvd == spec->rebuild_replica_count) {
```

Comment:
status response which are NOT OK, and, also, connection breaks after sending 'prepare_for_rebuild' command need to be handled.
Connection break kind of errors need to be handled in empty_mgmt_q_of_replica()

Reply:
Status NOT_OK need to be handled, I agree, but connection break be handled automatically when we call handle_mgmt_conn_error/handle_data_conn_error which I think bound to be called when there is problem w.r.t either connection.